### PR TITLE
Update normalising method in oneshot classifier

### PR DESCRIPTION
**Pull request by sachinruk**
Hi, I know I raised this issue before here: https://github.com/huggingface/transformers/pull/5760#issuecomment-673840015 but I do think that it is worth taking a second look atleast.

So my main concern is, since we are dealing with logits, currently all that happens is to take the softmax over the entailment logits. This does guarantee summing to one, however, it doesn't care for the scale of the logits. For example suppose the logits for two possible classes for a given sentence came out as:
```
[[1000, 10, 10],
 [0.1, 0.1, 0.9]]
```
at the current method would give class1 the higher probability, even though, it can clearly be seen that class1 is a contradiction (since it is 100x contradiction in the log scale), and similarly class2 is clearly an entailment.

Now I do understand that this is an extreme case, but just to cover the bases, what I propose is that:
1. We softmax across every single sentence-class pair to get everything in the same scale.
2. Get a probability measurement over entailment by simply dividing by sum of probabilities. As seen [here](https://github.com/huggingface/transformers/pull/5760#issuecomment-673840015 ) the numbers that we get are different but only slightly for the example shown.

I do apologise about raising this again, just simply trying to help.

### CODE REVIEW

**Comment by LysandreJik** [comment]
Pinging @joeddav

**Comment by joeddav** [comment]
Hey @sachinruk, thanks for taking the time to contribute 🤗

I take your point and the case that you've described can certainly happen with the current method, but I maintain what I said in the other thread that there's not really a single correct way of doing this. We're highjacking the outputs of a model trained on a completely different distribution (NLI data) for own own purposes, so it's just a game of figuring out what makes intuitive sense and what empirically works the best.

I ran a quick benchmark on the AG's News topic classification dataset comparing the current method with the one you've proposed, and they performed similarly. I got a weighted F1 of ~70 with the entailment-only method used in the pipeline and around 68 with the method you've proposed.

If you can show that your proposed method empirically does significantly better, we could look into changing it or adding it as an additional kwarg to specify the method. But as is, I don't think it makes sense to change the behavior of the pipeline under people's feet.
